gateway_lambda/index.ts

```diff
@@ -0,0 +1,146 @@
+import { APIGatewayEvent, APIGatewayProxyResult } from 'aws-lambda';
+import * as Sentry from '@sentry/serverless';
+import config from './config';
+import { validate } from './jwt';
+import { sendMessage } from './sqs';
+import { SqsEvent } from './types';
+
+Sentry.AWSLambda.init({
+  dsn: config.sentry.dsn,
+  release: config.sentry.release,
+  environment: config.environment,
+});
+
+/**
+ * Format the response data
+ * Source: https://docs.aws.amazon.com/lambda/latest/dg/services-apigateway.html#apigateway-types-transforms
+ * @param statusCode
+ * @param message
+ * @param isError
+ */
+export function formatResponse(
+  statusCode,
+  message: string,
+  isError?: boolean
+): APIGatewayProxyResult {
+  return {
+    statusCode,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      statusCode,
+      [isError ? 'error' : 'message']: message,
+    }),
+  };
+}
+
+/**
+ * Generate the data for the SQS messages
+ * @param data
+ */
+export function generateEvents(data): SqsEvent[] {
+  const userId = data.payload.sub;
+  const events = data.payload.events;
+  const allowedEvents = config.fxa.allowedEvents;
+
+  return Object.keys(events)
+    .filter((event) => Object.keys(allowedEvents).includes(event))
+    .map((event) => ({
+      user_id: userId,
+      event: allowedEvents[event],
+      timestamp: Math.round(new Date().getTime() / 1000),
+    }));
+}
+
+/**
+ * Create the message for a successful execution
+ * If messages fail, update message to point to
+ * where the logs are located.
+ * @param totalEvents
+ * @param failedEvents
+ */
+export function createSuccessResponseMessage(
+  totalEvents: number,
+  failedEvents: number
+): string {
+  const numberOfSuccessfulEvents = totalEvents - failedEvents;
+  let message = `Successfully sent ${numberOfSuccessfulEvents} out of ${totalEvents} events to SQS.`;
+  if (totalEvents > numberOfSuccessfulEvents) {
+    message +=
+      ' Review cloudwatch and sentry logs for information about failed events.';
```

Comment:
I started adding runbooks for the parser. Do you think you could add a placeholder for this one and fill in some guestimates similar to what I did https://getpocket.atlassian.net/l/c/rCitRCB2 && https://getpocket.atlassian.net/l/c/YffJPxpG

Reply:
and maybe add `expected ${totalEvents} received ${numberOfSuccessfulEvents}` or something like that to the message?

Reply:
@bkochendorfer When some messages are not sent the full message is `Successfully sent ${numberOfSuccessfulEvents} out of ${totalEvents} events to SQS. Review cloudwatch and sentry logs for information about failed events.`. Does that work or should I reword it?

Reply:
I added a [runbook](https://getpocket.atlassian.net/l/c/ewdAjX9c), not with that many guestimates, just mostly copied what you had. Let me know if that works for now

Reply:
Oh I missed the `+` there, thanks @kkyeboah :facepalm: